www/android/geolocation.js

```diff
@@ -38,21 +39,29 @@ module.exports = {
     },
 
     watchPosition: function(success, error, args) {
+        var pluginWatchId = utils.createUUID();
+
         var win = function() {
             var geo = cordova.require('cordova/modulemapper').getOriginalSymbol(window, 'navigator.geolocation');
-            geo.watchPosition(success, error, {
+            pluginToNativeWatchMap[pluginWatchId] = geo.watchPosition(success, error, {
                 enableHighAccuracy: args[1]
             });
         };
         exec(win, error, "Geolocation", "getPermission", []);
+
+        return pluginWatchId;
     },
 
-    clearWatch: function(success, error, args) {
+    clearWatch: function(pluginWatchId) {
         var win = function() {
-          var geo = cordova.require('cordova/modulemapper').getOriginalSymbol(window, 'navigator.geolocation');
-          geo.clearWatch(args[0]);
+            var nativeWatchId = pluginToNativeWatchMap[pluginWatchId];
```

Comment:
I tested this case and incorrect value (which does not exist or undefined) was ignored w/o any error or exception